File: task 2/serverless-email-api/test-api.js
const fetch = require('node-fetch');

const API_URL = 'http://localhost:3000/dev/send-email';

// Test data
const testEmail = {
  receiver_email: 'recipient@example.com', // Change this to your test email
  subject: 'Test Email from Serverless API',
  body_text: 'This is a test email sent from the serverless email API.',
};

async function testSendEmail() {
  try {
    console.log('Sending test email...');
    console.log('Payload:', JSON.stringify(testEmail, null, 2));

    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(testEmail),
    });

    const data = await response.json();

    console.log('\n--- Response ---');
    console.log('Status:', response.status);
    console.log('Data:', JSON.stringify(data, null, 2));

  } catch (error) {
    console.error('Error:', error.message);
  }
}

// Test missing fields
async function testMissingFields() {
  try {
    console.log('\n\nTesting missing fields...');
    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        receiver_email: 'test@example.com',
      }),
    });

    const data = await response.json();
    console.log('Status:', response.status);
    console.log('Data:', JSON.stringify(data, null, 2));

  } catch (error) {
    console.error('Error:', error.message);
  }
}

// Test invalid email
async function testInvalidEmail() {
  try {
    console.log('\n\nTesting invalid email...');
    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        receiver_email: 'invalid-email',
        subject: 'Test',
        body_text: 'Test body',
      }),
    });

    const data = await response.json();
    console.log('Status:', response.status);
    console.log('Data:', JSON.stringify(data, null, 2));

  } catch (error) {
    console.error('Error:', error.message);
  }
}

// Run all tests
async function runTests() {
  await testSendEmail();
  await testMissingFields();
  await testInvalidEmail();
}

runTests();

File: task 2/serverless-email-api/test_api.py
import requests
import json

API_URL = 'http://localhost:3000/dev/send-email'

def test_valid_email():
    """Test sending valid email"""
    print("Test 1: Valid email send")
    payload = {
        'receiver_email': 'your-email@gmail.com',  # CHANGE THIS
        'subject': 'Test Email from Python Serverless',
        'body_text': 'This is a test email. If you receive this, the API works!'
    }
    
    try:
        response = requests.post(API_URL, json=payload)
        print(f"Status: {response.status_code}")
        print(f"Response: {json.dumps(response.json(), indent=2)}\n")
    except Exception as e:
        print(f"Error: {e}\n")

def test_missing_field():
    """Test missing required field"""
    print("Test 2: Missing field (should return 400)")
    payload = {
        'receiver_email': 'test@example.com',
        'subject': 'Test'
        # Missing body_text
    }
    
    try:
        response = requests.post(API_URL, json=payload)
        print(f"Status: {response.status_code}")
        print(f"Response: {json.dumps(response.json(), indent=2)}\n")
    except Exception as e:
        print(f"Error: {e}\n")

def test_invalid_email():
    """Test invalid email format"""
    print("Test 3: Invalid email format (should return 400)")
    payload = {
        'receiver_email': 'not-an-email',
        'subject': 'Test',
        'body_text': 'Test body'
    }
    
    try:
        response = requests.post(API_URL, json=payload)
        print(f"Status: {response.status_code}")
        print(f"Response: {json.dumps(response.json(), indent=2)}\n")
    except Exception as e:
        print(f"Error: {e}\n")

def test_empty_string():
    """Test empty string field"""
    print("Test 4: Empty string (should return 400)")
    payload = {
        'receiver_email': 'test@example.com',
        'subject': '',
        'body_text': 'Test'
    }
    
    try:
        response = requests.post(API_URL, json=payload)
        print(f"Status: {response.status_code}")
        print(f"Response: {json.dumps(response.json(), indent=2)}\n")
    except Exception as e:
        print(f"Error: {e}\n")

if __name__ == '__main__':
    print("="*60)
    print("API TESTING")
    print("="*60 + "\n")
    
    test_valid_email()
    test_missing_field()
    test_invalid_email()
    test_empty_string()
    
    print("="*60)
    print("TESTS COMPLETE")
    print("="*60)